pkg/interface/chat/src/js/components/lib/sidebar-item.js

```diff
@@ -25,14 +25,26 @@ export class SidebarItem extends Component {
   }
 
   getTimeSinceNewestMessage() {
-    return !!this.props.wen ?
-      moment.unix(this.props.wen / 1000).from(moment.utc())
+    return !!this.props.when ?
+      moment.unix(this.props.when / 1000).from(moment.utc())
       : '';
   }
 
   onClick() {
     const { props } = this;
-    props.history.push('/~chat/' + props.cir);
+    props.history.push('/~chat/room' + props.box);
+  }
+
+  getLetter(lett) {
+    if ('text' in lett) {
+      return lett.text; 
+    } else if ('url' in lett) {
+      return lett.url;
+    } else if ('code' in lett) {
+      return lett.code.expression;
+    } else {
+      return '';
+    }
```

Comment:
A sequence of ternary operators might be useful here:

```
  'text' in lett
? lett.text
: 'url' in lett
? lett.url
[..]
```

Reply:
I don’t really like the style of using nested ternaries in JS. I generally go for my JS to be “explicit, needlessly verbose, and dumb as bricks”

Reply:
That's fine too.  I like the ternaries because you get a single expression, rather than a bunch of statements, but I get the motivation behind the alternative.